components/org.wso2.carbon.identity.query.saml/src/main/java/org/wso2/carbon/identity/query/saml/handler/SAMLAssertionFinderImpl.java

```diff
@@ -0,0 +1,155 @@
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied. See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package org.wso2.carbon.identity.query.saml.handler;
+
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.opensaml.saml.saml2.core.Assertion;
+import org.wso2.carbon.identity.core.persistence.JDBCPersistenceManager;
+import org.wso2.carbon.identity.core.util.IdentityDatabaseUtil;
+import org.wso2.carbon.identity.query.saml.util.SAMLQueryRequestUtil;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Implementation class to find assertions dynamically from assertion stores
+ *
+ * @see SAMLAssertionFinder
+ */
+public class SAMLAssertionFinderImpl implements SAMLAssertionFinder {
+    private static final Log log = LogFactory.getLog(SAMLAssertionFinderImpl.class);
+
+    /**
+     * This method is used to initialize handler
+     */
+    public void init() {
+
+    }
+
+    /**
+     * Method to finf assertions from assertion stores by assertion is
+     *
+     * @param id This is the unique Assertion ID
+     * @return Assertion This returns assertion with given assertion id
+     */
+    public Assertion findByID(String id) {
+        Assertion assertion = null;
+        if (id.length() > 0 && id != null) {
+            assertion = readAssertion(id);
+            return assertion;
+
+        } else {
+            log.error("Assertion ID field is Empty");
+
+        }
+
+        return assertion;
+    }
+
+    /**
+     * This method is used to search messages from assertion stores by subject of the assertion
+     *
+     * @param subject This is full qualified subject of the request
+     * @return List This returns a list of assertions
+     */
+    public List<Assertion> findBySubject(String subject) {
+        List<Assertion> assertions = new ArrayList<Assertion>();
+        Connection connection = null;
+        PreparedStatement preparedStatement = null;
+        ResultSet resultSet = null;
+        try {
+            connection = JDBCPersistenceManager.getInstance().getDBConnection();
+            preparedStatement = connection.prepareStatement("SELECT SAML2_ASSERTION FROM IDN_SAML2_ASSERTION_STORE WHERE SAML2_SUBJECT= ?");
+            preparedStatement.setString(1, subject);
+            resultSet = preparedStatement.executeQuery();
+            int index = 1;
+            while (resultSet.next()) {
+                String assertionString = resultSet.getString(index);
+                try {
+                    assertions.add((Assertion) SAMLQueryRequestUtil.unmarshall(assertionString));
+                    log.debug("Assertions retrieved from database");
```

Comment:
add some context info to the debug message,
eg: "Assertions retrieved from database for subject : " + subject